### PR TITLE
feat: disable default props

### PR DESCRIPTION
**Pull request by p-spacek**
### What does this PR do?
Add config to exclude not required default props from completion.
fix: required prop with default value or with const

The reason why users appreciated this feature is that users don't have to specify default properties. Only if the default value needs to be changed, it makes sense to add them (but not all of them automatically).

### What issues does this PR fix or reference?
no ref

### Is it tested? How?
add unit tests

### CODE REVIEW

**Comment by evidolob** [comment]
@p-spacek It would be nice if you add description for new settings item in to [README.md](https://github.com/redhat-developer/yaml-language-server#language-server-settings)

File: test/autoCompletion.test.ts [hunk 1]
```diff
@@ -386,6 +386,62 @@ describe('Auto Completion Tests', () => {
...
+      it('Autocomplete without default value - not required', async () => {
+        const languageSettingsSetup = new ServiceSetup().withCompletion();
+        languageSettingsSetup.languageSettings.disableDefaultProperties = true;
+        languageService.configure(languageSettingsSetup.languageSettings);
+        languageService.addSchema(SCHEMA_ID, {
+          type: 'object',
+          properties: {
+            scripts: {
+              type: 'object',
+              properties: {
```
Comment (gorkem) [inline test/autoCompletion.test.ts:398]: Can we possibly have a second non-string property added to this test case?

File: test/autoCompletion.test.ts [hunk 2]
```diff
@@ -386,6 +386,62 @@ describe('Auto Completion Tests', () => {
...
+      it('Autocomplete without default value - required', async () => {
+        const languageSettingsSetup = new ServiceSetup().withCompletion();
+        languageSettingsSetup.languageSettings.disableDefaultProperties = true;
+        languageService.configure(languageSettingsSetup.languageSettings);
+        languageService.addSchema(SCHEMA_ID, {
+          type: 'object',
+          properties: {
+            scripts: {
+              type: 'object',
+              properties: {
```
Comment (gorkem) [inline test/autoCompletion.test.ts:425]: Can we have a non-string property added?

**Review by gorkem — CHANGES_REQUESTED** [review]
A small improvement to the test cases but otherwise LGTM. Is there a corresponding PR needed for the vscode-yaml repo?

File: test/autoCompletion.test.ts (hunk 1 above)
Comment (p-spacek) [reply test/autoCompletion.test.ts:398]: I'm sorry for the longer delay...
I modified the UT. Have a look if I understood correctly your note, please.
vscode-yaml PR is here https://github.com/redhat-developer/vscode-yaml/pull/667

**Review by gorkem — APPROVED** [review]
This looks ready to go. I am not sure why CI is unhappy with the new test though

**Comment by evidolob** [comment]
@p-spacek Test fails because of https://github.com/redhat-developer/yaml-language-server/pull/609 , could you update this?

**Comment by p-spacek** [comment]
> @p-spacek Test fails because of #609 , could you update this?

oh sure.
updated

**Review by evidolob — APPROVED** [review]

**Pull request merged**